guides/V1_ActionsEvents.md

```diff
@@ -0,0 +1,183 @@
+
+
+## Adding a client action handler callback
+
+To add a Client Action Handler for your App, you need to;
+
+1. Go to Your Apps and choose the App.
+
+![Your Apps](../images/YourApps1.png)
+
+2. Click Add an Action.
+
+![My App View](../images/YourApps2.png)
+
+3. Then click Add a configuration handler.
+
+![AppConfig Usage Screenshot](../images/AddAction1.png)
+
+4. Add your URL and select at least one of *Space configuration* or *Team configuration*.
+
+![AppConfig Usage Screenshot](../images/AddAction2.png)
+
+5. click Add configuration handler.
+
+![AppConfig Usage Screenshot](../images/AddAction3.png)
+
+6. See a confirmation of the addition.
+
+![AppConfig Usage Screenshot](../images/AddAction4.png)
+
+7. You will see your Client Action Handler...
+
+![AppConfig Usage Screenshot](../images/AddAction5.png)
+
+8. ...which you can Edit or Delete in the future.
+
+![AppConfig Usage Screenshot](../images/AddAction6.png)
+
+## Things to consider when you have an explicit action handler
+
+It's important to think about these things when adding an explicit action handler to your App.
+
+ - The explicit action can be requested by anyone, so the caller and the provided parameters must be verified.
+
+ - If your explicit action requires knowledge of the user, then you must obtain the user id and name also to establish a link between the user's ID and the user's name.
+
+ - If your explicit action takes into account something to do with the space, then you should display the space name to help the caller verify that it is indeed the space for which this action should take place.
+
+ - You should display a continue to space button to complete the process.
+
+ Let's walk through these.
+
+### Protect the explicit action endpoint
+
+Making an endpoint publicly available has security risks; for example, your endpoint could receive requests from malicious sources. To minimize risk:
+- Be prepared for <a href="https://en.wikipedia.org/wiki/Network_security#Types_of_Attacks" target="_blank">network attacks</a> and ensure that appropriate counter-measures are in place.
+- Do not assume that the request or its parameters are trustworthy until you have verified the call.
+
+
+### Verify the call
+
+Use the url parameter `actionToken` and utilize Watson Work Services to verify the call.
+Authorize your app [as an App](../references/V1_oauth_token_client_credentials.yml) and access the action data with the `actionToken` - see [Get Action Context](./V1_GetActionContext.md).
+If the access to the action data is successful, then you can be sure that the call has been triggered by Watson Workspace.
+You will also get the action context data, which may include things like `spaceId` and `userId`.  `spaceId` is the unique ID of the space for which the action has been triggered.  While `userId` is the unique ID of the user who triggers the action.
```

Comment:
this sentence I think is left over of the specific data for app config in space.  It should be removed as likely already documented separately.